File: C++/M01_ProgrammingBasics/L05_WhileLoop/Exercises/Solutions/P04_Walking.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {

	string command;
	int goal = 10000;
	int step;

	while (goal > 0) {
		getline(cin >> ws, command);
		if (command == "Going home") {
			cin >> command;
			step = stoi(command);
			goal -= step;
			if (goal > 0) {
				cout << goal << " more steps to reach goal." << endl;
			}
			break;
		}
		step = stoi(command);
		goal -= step;
	}
	if (goal <= 0) {
		cout << "Goal reached! Good job!" << endl;
	}

	return 0;
}

File: Python/M01_ProgrammingBasics/L05_WhileLoop/Exercises/Solutions/P04_Walking.py
target_steps = 10000
steps_done = 0

while True:
    line = input()

    if line == "Going home":
        home_steps = int(input())
        steps_done = steps_done + home_steps
        if steps_done >= target_steps:
            print("Goal reached! Good job!")
            print(f"{steps_done - target_steps} steps over the goal!")
            break
        else:
            print(f"{target_steps - steps_done} more steps to reach goal.")
            break

    steps = int(line)
    steps_done += steps

    if steps_done >= target_steps:
        print("Goal reached! Good job!")
        print(f"{steps_done - target_steps} steps over the goal!")
        break

